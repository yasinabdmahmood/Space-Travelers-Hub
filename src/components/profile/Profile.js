import { useSelector } from 'react-redux';
import style from './profile.module.css';

function Profile() {
  const rockets = useSelector((state) => state.rockets);
  const missions = useSelector((state) => state.missions);
  return (
    <div className={style.container} data-testid="main-container">
      <div className={style.missions}>
        <h1>My Missions</h1>
        <ul className={style['rocket-list']}>
          {missions.filter((el) => el.joined).map((el) => (
            <li key={el.id} className={style['rocket-item']}>
              {el.mission_name}
            </li>
          ))}
        </ul>
      </div>
      <div className={style.rockets}>
        <h1>My Rockets</h1>
        <ul className={style['rocket-list']}>
          {
          rockets.filter((el) => el.reserved).map((el) => (
            <li key={el.id}>
              {el.rocket_name}
            </li>
          ))
         }
        </ul>
      </div>
    </div>
  );
}

export default Profile;
