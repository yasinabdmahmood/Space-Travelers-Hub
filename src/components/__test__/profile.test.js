import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Profile from '../profile/Profile';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if the component has changed', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Profile />
    </Provider>,

  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('the component should render the main container', () => {
  render(
    <Provider store={store}>
      <Profile />
    </Provider>,
  );

  const container = screen.getByTestId('main-container');

  expect(container).toBeInTheDocument();
});
