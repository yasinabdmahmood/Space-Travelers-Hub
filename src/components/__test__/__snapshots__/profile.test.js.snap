// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component has changed 1`] = `
<div
  className="container"
  data-testid="main-container"
>
  <div
    className="missions"
  >
    <h1>
      My Missions
    </h1>
    <ul
      className="rocket-list"
    />
  </div>
  <div
    className="rockets"
  >
    <h1>
      My Rockets
    </h1>
    <ul
      className="rocket-list"
    />
  </div>
</div>
`;
